src/librustc_metadata/index_builder.rs

```diff
@@ -8,6 +8,60 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
+//! Builder types for generating the "item data" section of the
+//! metadata. This section winds up looking like this:
+//!
+//! ```
+//! <common::data> // big list of item-like things...
+//!    <common::data_item> // ...for most def-ids, there is an entry.
+//!    </common::data_item>
+//! </common::data>
+//! ```
+//!
+//! As we generate this listing, we collect the offset of each
+//! `data_item` entry and store it in an index. Then, when we load the
+//! metadata, we can skip right to the metadata for a particular item.
+//!
+//! In addition to the offset, we need to track the data that was used
+//! to generate the contents of each `data_item`. This is so that we
+//! can figure out which HIR nodes contributors to that data for
+//! incremental compilation purposes.
+//!
+//! The `IndexBuilder` facilitates with both of these. It is created
```

Comment:
For what it's worth, one can write either "facilitates with X" or "facilitates X", as far as I know (maybe some style guide says no)...but I agree the latter is prob better. :)